File: Java/src/main/java/com/gildedrose/ConjuredUpdater.java
package com.gildedrose;

public class ConjuredUpdater extends StandardItemUpdater {

    @Override
    int getUpdateValue(final Item item) {
        if (item.sellIn < 0) {
            return DEGRADE_TWICE_AS_FAST * 2;
        } else {
            return DEGRADE_TWICE_AS_FAST;
        }
    }
}

File: js-jest/src/item_updaters/conjured_updater.js
const { ItemUpdater } = require("../gilded_rose");

class ConjuredUpdater extends ItemUpdater {
    updateQuality() {

        this.qualityChangeFactor = 2;

        //The Quality of an item is never more than 50
        if (this.item.quality > 50) {
            this.item.quality = 50;
        }

        if (this.item.quality > 0 && this.item.quality < 50) {
            this.item.quality = this.item.quality - this.qualityChangeFactor;
        }
        return this.item;
    }
}

module.exports = {
    ConjuredUpdater
}